File: python/Strings/day3/isomorphicStrings.py
def is_isomorphic(s, t):
    if len(s) != len(t):
        return False
    
    map_st = {}
    map_ts = {}

    for char_s, char_t in zip(s, t):
        # Check mapping s -> t...
        if char_s in map_st:
            if map_st[char_s] != char_t:
                return False
            
        else:
            map_st[char_s] = char_t


        # Check mapping t -> s...
        if char_t in map_ts:
            if map_ts[char_t] != char_s:
                return False
        
        else:
            map_ts[char_t] = char_s

    return True


if __name__ == "__main__":
    s = input("Enter first string: ")
    t = input("Enter second string: ")

    result = is_isomorphic(s, t)

    if result:
        print("Result: The strings ARE isomorphic.")
    else:
        print("Result: The strings are NOT isomorphic.")

File: java/Strings/Day3/IsomorphicStrings.java
package Day3;
import java.util.*;

public class IsomorphicStrings {
    public boolean isIsomorphic(String s, String t) {
        if (s.length() != t.length()) return false;

        HashMap<Character, Character> mapST = new HashMap<>();
        HashMap<Character, Character> mapTS = new HashMap<>();


        for (int i = 0; i < s.length(); i++) {
            char charS = s.charAt(i);
            char charT = s.charAt(i);


            // Check mapping S -> T...
            if (mapST.containsKey(charS)) {
                if (mapST.get(charS) != charT) return false;
            }
            else {
                mapST.put(charS, charT);
            }

            // Check mapping T -> S...
            if (mapTS.containsKey(charT)) {
                if (mapTS.get(charT) != charS) return false;
            }
            else {
                mapTS.put(charT, charS);
            }
        }

        return true;
    }


    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter first string: ");
        String s = sc.nextLine();

        System.out.print("Enter second string: ");
        String t = sc.nextLine();

        IsomorphicStrings obj = new IsomorphicStrings();
        boolean result = obj.isIsomorphic(s, t);

        if (result) {
            System.out.println("Result: The strings ARE isomorphic.");
        } else {
            System.out.println("Result: The strings are NOT isomorphic.");
        }

        sc.close();
    }
}